feat: implement best selling and most popular products sections

MohamedAbdelhakem17 opened the pull request.

Commit:
feat(best-selling): create best selling section and support translation

Commit:
feat(home): integrate API to fetch and display home data including best selling and most popular products, add getOccasions function, and define Occasions types

Shamsmedhat commented on src/app/[locale]/(auth)/login/_components/login-form.tsx:
```diff
@@ -9,15 +10,15 @@ import {
+
+import { Button } from '@/components/ui/button';
+import { useTranslations } from 'next-intl';
+import Link from 'next/link';
```
We import all navigation methods and properties from next intl.

Shamsmedhat commented on src/app/[locale]/(auth)/register/_components/register-form.tsx:
```diff
@@ -1,29 +1,28 @@
-import FormInput from '@/components/shared/form-input';
+import { useRegister } from '../_hooks/use-register';
 import { PasswordInput } from './register-form-password-input';
 // import { PasswordInput } from '@/components/shared/password-input';
```
Please make sure not to leave any commented-out code when pushing your changes unless it’s intentionally kept for future use.

Shamsmedhat commented on src/app/[locale]/(auth)/register/_components/register-form.tsx:
```diff
@@ -1,29 +1,28 @@
 // import { PasswordInput } from '@/components/shared/password-input';
 
 export default function RegisterForm() {
   //react query mutation hook
```
Our comments approach.
```
export default function MyComponent() {

// Translation
// Navigation
// State
// Ref
// Context
// Hooks
// Queries
// Mutation
// Form & validation
// Variables => This is a flexible choice
// Functions
// Effects

return <><>
}
```

Shamsmedhat commented on src/app/[locale]/(public)/(home-page)/_components/best-selling/_components/best-selling-content.tsx:
Good job 👏👏

Shamsmedhat commented on src/app/[locale]/(public)/(home-page)/_components/best-selling/_components/product-slider-list.tsx:
```diff
@@ -0,0 +1,52 @@
+  products: MappingProductType[];
+}) {
+  const plugin = useRef(
+    typeof window !== 'undefined'
```
This is not correct approach.

[Link](https://nextjs.org/docs/messages/react-hydration-error)

Shamsmedhat commented on src/app/[locale]/(public)/(home-page)/_components/best-selling/_components/product-slider-list.tsx:
Need comments on this component.

Shamsmedhat commented on src/app/[locale]/(public)/(home-page)/_components/best-selling/index.tsx:
```diff
@@ -0,0 +1,16 @@
+      </div>
+
+      {/* Slider */}
+      <ProductSlider />
```
Where is the suspense component ?.
When you fetch data in a server component (server-side), you should wrap the component that performs the fetch in a `<Suspense>` component with a proper fallback (e.g., skeleton, loading UI). Also, ensure that the wrapped component is marked as async so that `<Suspense>` works correctly and displays the fallback while waiting for the promise to resolve.

Shamsmedhat commented on src/app/[locale]/(public)/(home-page)/_components/most-popular/_components/most-popular-header.tsx:
```diff
@@ -0,0 +1,23 @@
+
+  // Query
+  const response: GetOccasionsTypeResponse = await getOccasions();
+  const occasions = 'error' in response ? [] : response.occasions;
```
When we handle error we need to know what is the error before we return just an empty array.

Shamsmedhat commented on src/app/[locale]/(public)/(home-page)/_components/most-popular/_components/occasions-list.tsx:
```diff
@@ -0,0 +1,35 @@
+  const searchParams = useSearchParams();
+  const handleSearchFilter = useSearchFilter({ scroll: false });
+
+  console.log(searchParams);
```
REmove logs.

Shamsmedhat commented on src/app/[locale]/(public)/(home-page)/_components/most-popular/_components/occasions-list.tsx:
Remove comments.

Shamsmedhat commented on src/app/[locale]/(public)/(home-page)/_components/most-popular/index.tsx:
```diff
@@ -0,0 +1,49 @@
+
+        {/* Products */}
+        <Suspense
+          fallback={Array.from({ length: 6 }).map((_, index) => (
```
Good job 👏, however better to create a component for skeleton.

Shamsmedhat commented on src/app/[locale]/(public)/(home-page)/_components/most-popular/index.tsx:
```diff
@@ -0,0 +1,49 @@
+import { cn } from '@/lib/utils/utils';
+import { MoveRight } from 'lucide-react';
+import { getLocale, getTranslations } from 'next-intl/server';
+import Link from 'next/link';
```
This should be from next intl not next/navigation.

Shamsmedhat reviewed (commented):
Good job overall 👏

Commit:
refactor(home): clean up code and improve loading UX

- Removed unnecessary console logs and comments
- Added meaningful comments to describe code sections
- Implemented skeletons for loading states
- Moved skeleton logic into a separate component for better structure and reusability

MohamedAbdelhakem17 committed:
Merge branch 'dev' into feat/best-selling-and-most-popular
